File: push/src/main/java/com/xiaomi/push/service/XMPushServicePingSupport.java
package com.xiaomi.push.service;

import java.util.ArrayList;
import java.util.Iterator;

final class XMPushServicePingSupport {
    private XMPushServicePingSupport() {
    }

    static void onPong(ArrayList<PingCallBack> pingCallBacks) {
        Iterator it = new ArrayList(pingCallBacks).iterator();
        while (it.hasNext()) {
            ((PingCallBack) it.next()).pingFollowUpAction();
        }
    }
}

File: legacy/src/main/java/com/xiaomi/push/service/XMPushServicePingSupport.kt
package com.xiaomi.push.service
import io.github.magisk317.mipush.protocol.model.*

object XMPushServicePingSupport {
    @JvmStatic
    fun onPong(pingCallBacks: ArrayList<PingCallBack>) {
        ArrayList(pingCallBacks).forEach { it.pingFollowUpAction() }
    }
}
